programs/github

```diff
@@ -59,4 +132,26 @@ if action == 'clone' then
 	term.scroll(1)
 	return
 end
-print(usage)
+
+if action == 'auth' then
+	if flags.delete then
+		github.Auth.delete(flags.delete)
+		print(('Github authentication for %s successfully deleted.').format(flags.delete))
+	else
+		local user, token = args[1], args[2]
+		if not user or not token then return printError("No user or token specified.") end
+		local auth = github.Auth('basic', user, token)
+		auth.save()
+		print(('Github user %s successfully saved.').format(auth.user))
+	end
+
+	return
+end
+
+if not validActions[action] then
+	for k, v in pairs(validActions) do
+		print(('%s usage: %s').format(v.name, v.usage))
```

Comment:
Terminal width is a premium here, so I'm -1 on printing anything other than the usage string